qiskit/transpiler/passes/optimization/consolidate_blocks.py

```diff
@@ -113,9 +115,18 @@ def run(self, dag):
                     or ((self.basis_gates is not None) and outside_basis)
                     or ((self.target is not None) and outside_basis)
                 ):
-                    dag.replace_block_with_op(block, unitary, block_index_map, cycle_check=False)
+                    identity = np.eye(2**unitary.num_qubits)
+                    if np.allclose(identity, unitary.to_matrix()):
+                        for node in block:
+                            dag.remove_op_node(node)
+                    else:
+                        dag.replace_block_with_op(
+                            block, unitary, block_index_map, cycle_check=False
+                        )
         # If 1q runs are collected before consolidate those too
         runs = self.property_set["run_list"] or []
+        if runs:
+            identity_1q = np.eye(2)
```

Comment:
minor style notes: I've never been a fan of conditional name binding for a name that's used later.  Constructing a single 2x2 identity matrix is a tiny cost - maybe we can just create this unconditionally?

Reply:
Yeah I guess I'm just micro-optimizing a bit too much. I'll drop the if here.

Reply:
Fixed in: https://github.com/Qiskit/qiskit-terra/pull/9222/commits/ad5e44091ffd3fb4be41c5a237e74ef7e21715f9